src/io/io.cpp

```diff
@@ -1603,37 +1617,46 @@ void Grid3D::Write_Projection_HDF5(hid_t file_id)
     }
 
     // Copy the xz density and temperature projections to the memory buffer
-    for (k = 0; k < H.nz_real; k++) {
-      for (i = 0; i < H.nx_real; i++) {
+    for (int k = 0; k < H.nz_real; k++) {
+      for (int i = 0; i < H.nx_real; i++) {
         dxz = 0;
         Txz = 0;
   #ifdef DUST
         dust_xz = 0;
   #endif
         // for each xz element, sum over the y column
-        for (j = 0; j < H.ny_real; j++) {
-          id = (i + H.n_ghost) + (j + H.n_ghost) * H.nx + (k + H.n_ghost) * H.nx * H.ny;
+        for (int j = 0; j < H.ny_real; j++) {
+          int const xid = i + H.n_ghost;
+          int const yid = j + H.n_ghost;
+          int const zid = k + H.n_ghost;
+          int const id  = cuda_utilities::compute1DIndex(xid, yid, zid, H.nx, H.ny);
           // sum density
           dxz += C.density[id] * H.dy;
   #ifdef DUST
           dust_xz += C.dust_density[id] * H.dy;
   #endif
           // calculate number density
-          n = C.density[id] * DENSITY_UNIT / (mu * MP);
-  // calculate temperature
-  #ifndef DE
-          Real mx = C.momentum_x[id];
-          Real my = C.momentum_y[id];
-          Real mz = C.momentum_z[id];
-          Real E  = C.Energy[id];
-          T       = (E - 0.5 * (mx * mx + my * my + mz * mz) / C.density[id]) * (gama - 1.0) * PRESSURE_UNIT / (n * KB);
-  #endif
   #ifdef DE
-          T = C.GasEnergy[id] * PRESSURE_UNIT * (gama - 1.0) / (n * KB);
-  #endif
-          Txz += T * C.density[id] * H.dy;
+          Real const T = hydro_utilities::Calc_Temp_DE(C.density[id], C.GasEnergy[id], gama, n);
```

Comment:
Has n been calculated? Looks like maybe the calculation was accidentally deleted in this loop.

Reply:
Just added it back.